cli/npm/semver/mod.rs

```diff
@@ -0,0 +1,997 @@
+      ("~2.4", ">=2.4.0 <2.5.0-0"),
+      ("~>3.2.1", ">=3.2.1 <3.3.0-0"),
+      ("~1", ">=1.0.0 <2.0.0-0"),
+      ("~>1", ">=1.0.0 <2.0.0-0"),
+      ("~> 1", ">=1.0.0 <2.0.0-0"),
+      ("~1.0", ">=1.0.0 <1.1.0-0"),
+      ("~ 1.0", ">=1.0.0 <1.1.0-0"),
+      ("^0", "<1.0.0-0"),
+      ("^ 1", ">=1.0.0 <2.0.0-0"),
+      ("^0.1", ">=0.1.0 <0.2.0-0"),
+      ("^1.0", ">=1.0.0 <2.0.0-0"),
+      ("^1.2", ">=1.2.0 <2.0.0-0"),
+      ("^0.0.1", ">=0.0.1 <0.0.2-0"),
+      ("^0.0.1-beta", ">=0.0.1-beta <0.0.2-0"),
+      ("^0.1.2", ">=0.1.2 <0.2.0-0"),
+      ("^1.2.3", ">=1.2.3 <2.0.0-0"),
+      ("^1.2.3-beta.4", ">=1.2.3-beta.4 <2.0.0-0"),
+      ("<1", "<1.0.0-0"),
+      ("< 1", "<1.0.0-0"),
+      (">=1", ">=1.0.0"),
+      (">= 1", ">=1.0.0"),
+      ("<1.2", "<1.2.0-0"),
+      ("< 1.2", "<1.2.0-0"),
+      ("1", ">=1.0.0 <2.0.0-0"),
+      ("^ 1.2 ^ 1", ">=1.2.0 <2.0.0-0 >=1.0.0"),
+      ("1.2 - 3.4.5", ">=1.2.0 <=3.4.5"),
+      ("1.2.3 - 3.4", ">=1.2.3 <3.5.0-0"),
+      ("1.2 - 3.4", ">=1.2.0 <3.5.0-0"),
+      (">1", ">=2.0.0"),
+      (">1.2", ">=1.3.0"),
+      (">X", "<0.0.0-0"),
+      ("<X", "<0.0.0-0"),
+      ("<x <* || >* 2.x", "<0.0.0-0"),
+      (">x 2.x || * || <x", "*"),
+      // (">01.02.03", ">1.2.3", true),
+      // (">01.02.03", null),
+      // ("~1.2.3beta", ">=1.2.3-beta <1.3.0-0", { loose: true }),
+      // ("~1.2.3beta", null),
+      // (">=09090", null),
+      // (">=09090", ">=9090.0.0", true),
+      // (">=09090-0", null, { includePrerelease: true }),
+      // (">=09090-0", null, { loose: true, includePrerelease: true }),
+    ];
+    for (range_text, expected) in fixtures {
+      let range = NpmVersionReq::parse(range_text).unwrap();
+      let expected_range = NpmVersionReq::parse(expected).unwrap();
+      assert_eq!(
+        range.range_set, expected_range.range_set,
+        "failed for {} and {}",
+        range_text, expected
+      );
+    }
+  }
+
+  #[test]
+  fn range_satisfies() {
+    // https://github.com/npm/node-semver/blob/4907647d169948a53156502867ed679268063a9f/test/fixtures/range-include.js
+    let fixtures = &[
+      ("1.0.0 - 2.0.0", "1.2.3"),
+      ("^1.2.3+build", "1.2.3"),
+      ("^1.2.3+build", "1.3.0"),
+      ("1.2.3-pre+asdf - 2.4.3-pre+asdf", "1.2.3"),
+      // ("1.2.3pre+asdf - 2.4.3-pre+asdf", "1.2.3", true),
+      // ("1.2.3-pre+asdf - 2.4.3pre+asdf", "1.2.3", true),
+      // ("1.2.3pre+asdf - 2.4.3pre+asdf", "1.2.3", true),
+      ("1.2.3-pre+asdf - 2.4.3-pre+asdf", "1.2.3-pre.2"),
+      ("1.2.3-pre+asdf - 2.4.3-pre+asdf", "2.4.3-alpha"),
+      ("1.2.3+asdf - 2.4.3+asdf", "1.2.3"),
+      ("1.0.0", "1.0.0"),
+      (">=*", "0.2.4"),
+      ("", "1.0.0"),
+      // ("*", "1.2.3", {}),
+      // ("*", "v1.2.3", { loose: 123 }),
+      // (">=1.0.0", "1.0.0", /asdf/),
+      // (">=1.0.0", "1.0.1", { loose: null }),
+      // (">=1.0.0", "1.1.0", { loose: 0 }),
+      // (">1.0.0", "1.0.1", { loose: undefined }),
+      (">1.0.0", "1.1.0"),
```

Comment:
Why are some of these commented out?

Reply:
I forgot to investigate these ones. Just did now and made the parsing looser.